cores/arduino/pins_arduino_digital.h

```diff
@@ -0,0 +1,555 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020, STMicroelectronics
+ * All rights reserved.
+ *
+ * This software component is licensed by ST under BSD 3-Clause license,
+ * the "License"; You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ *******************************************************************************
+ */
+#ifndef _PINS_ARDUINO_DIGITAL_H_
+#define _PINS_ARDUINO_DIGITAL_H_
+#include "variant.h"
+
+/* GPIO port (A to K + Z) * 16 pins: 192 */
+/* Arduino digital pins alias */
+#define D0       0
```

Comment:
AFAIK Using an enum or a macro does not change the issue if `Dx` is used in other code as it is global. I've changed to be inline with Ax definition and allow preprocor directive at build time.

Reply:
But for Ax, the `Ax` constants are *not* macros, only the (longer, thus less likely to cause problems) `PIN_Ax` constants are macros.

Consider a line of code like this:

```
void foo() {
  int D0 = 0;
}
```

With the changes in this commit (changing `D0` from an enum to a macro), that code no longer compiles, because the `D0` variable name is now expanded as a macro, yielding `int 0 = 0;`.

Reply:
@matthijskooijman  I've revert Dx to an enum.